src/Mailjet/Client.php

```diff
@@ -183,4 +216,70 @@ public function setSecureProtocol($bIsSecured)
 
         return false;
     }
+    
+    // TODO : make the next code more readable
+    
+    /**
+     * Temporary set the variables generating the url
+     * @param array $options    contain temporary modifications for the client
+     * @param array $resource   may contain the version linked to the ressource
+     */
+    private function setOptions($options, $resource)
+    {
+        if (!empty($options['version']) && is_string($options['version'])) {
```

Comment:
Array_key_exist only check if the key is set, not if the key is set and is not null. Also, it's slower than isset or empty.